L/LibCURL/build_tarballs.jl

```diff
@@ -40,8 +40,13 @@ elif [[ ${target} == *darwin* ]]; then
     # On Darwin, we need to use SecureTransport (native TLS library)
     FLAGS+=(--with-secure-transport)
 
-    # We need to explicitly request a higher `-mmacosx-version-min` here, so that it doesn't 
-    export CFLAGS=-mmacosx-version-min=10.11
+    # We need to explicitly request a higher `-mmacosx-version-min` here, so that it doesn't
+    # complain about: `Symbol not found: ___isOSVersionAtLeast`
+    if [[ "${target}" == aarch64* ]]; then
+        export CFLAGS=-mmacosx-version-min=11.0
+    else
+        export CFLAGS=-mmacosx-version-min=10.11
```

Comment:
Yeah, this is the kind of thing that I would like to support via the `os_version` key in the `Platform` object; it's why I would like to tie the `-mmacosx-version-min=` compiler flag to just `os_version(target)`, and we're a good part of the way there, but not quite yet.  What this will mean is that you'll be able to build multiple versions of the same package targeting different versions of the OS (to make use of internal APIs that aren't available on earlier versions, for instance) and that Julia will be able to select the appropriate one, or none at all if there are no compatibilties.